apis/root_handler.go

```diff
@@ -11,11 +14,17 @@ const (
 )
 
 type RootHandler struct {
+	Logger    logr.Logger
 	ServerURL string
 }
 
 func (h *RootHandler) RootGetHandler(w http.ResponseWriter, r *http.Request) {
-	body := `{"links":{"self":{"href":"` + h.ServerURL + `"},"bits_service":null,"cloud_controller_v2":null,"cloud_controller_v3":{"href":"` + h.ServerURL + `/v3","meta":{"version":"3.90.0"}},"network_policy_v0":null,"network_policy_v1":null,"login":null,"uaa":null,"credhub":null,"routing":null,"logging":null,"log_cache":null,"log_stream":null,"app_ssh":null}}`
+	body, err := json.Marshal(presenter.GetRootResponse(h.ServerURL))
+	if err != nil {
+		h.Logger.Error(err, "Failed to render response")
```

Comment:
We might want the error to be slightly more descriptive - if we see it in the logs, we may not know which response failed to render.

Reply:
I think it's not too bad since the logger has context about what handler we're in:

https://github.com/cloudfoundry/cf-k8s-api/pull/91/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R56

But in general I'm with you it would be nice to know what function we're in... does logging the error include the stack trace or no?

Reply:
They should, so this may not be needed, then.